src/internal/components/ScaledTextInput.tsx

```diff
@@ -0,0 +1,53 @@
+import { useEffect, useRef, useState } from 'react';
+import { cn } from '../../styles/theme';
+import type { TextInputReact } from './TextInput';
+import { TextInput } from './TextInput';
+
+export function ScaledTextInput(props: TextInputReact) {
+  const inputContainer = useRef<HTMLDivElement | null>(null);
+  const offscreenRef = useRef<HTMLSpanElement | null>(null);
+  const [containerScale, setContainerScale] = useState(1);
+
+  useEffect(() => {
+    if (props.value && offscreenRef.current && inputContainer.current) {
+      const containerWidth = inputContainer.current.offsetWidth;
+      const textWidth = offscreenRef.current.offsetWidth;
+      const scaleFactor = containerWidth / textWidth;
+
+      if (scaleFactor < 1) {
+        setContainerScale(scaleFactor);
+      } else {
+        setContainerScale(1);
+      }
+    } else {
+      setContainerScale(1);
+    }
+  }, [props.value]);
+
+  return (
+    <>
+      <div ref={inputContainer}>
+        <div
+          className="origin-[0_50%] overflow-hidden"
+          style={{
+            transform: `scale(${containerScale})`,
+            width: `${100 / containerScale}%`,
+          }}
```

Comment:
We should refrain from using inline-styling

Reply:
@cpcramer I think we are forced here to have the inline style. As this is a dynamic CSS story, which I don't think Tailwind has an easy solution for it.

@cpcramer this is also very similar on how CSS animation are built, where you have a basic class, but then use inline style to do crazy animaiton tricks.